### PR TITLE
Add doc to introduce how to use devfile on Z & Power.

### PR DESCRIPTION
**Pull request by zhengxiaomei123**
/kind documentation
[skip ci]

**What does this PR do / why we need it**:
To introduce how to use defiles on IBM Z & Power for clients.

### CODE REVIEW

File: chosen-docs/upstream/deploying-a-devfile-using-odo-on-IBM-Z-and-Power.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+# Page settings
+layout: default
+keywords:
+comments: false
+
+# Hero section
+title: Title here
+description: Description here
```
Comment (feloy) [inline chosen-docs/upstream/deploying-a-devfile-using-odo-on-IBM-Z-and-Power.md:9]: Please set Title and description

File: chosen-docs/upstream/deploying-a-devfile-using-odo-on-IBM-Z-and-Power.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# Micro navigation
+micro_nav: true
+
+# Page navigation
+page_nav:
+    prev:
+        content: Previous page
```
Comment (feloy) [inline chosen-docs/upstream/deploying-a-devfile-using-odo-on-IBM-Z-and-Power.md:17]: content should be "Deploying your first application using odo"
and url "/docs/deploying-a-devfile-using-odo'"

File: chosen-docs/upstream/deploying-a-devfile-using-odo-on-IBM-Z-and-Power.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        url: '/docs/previous-page'
+    next:
+        content: Next page
```
Comment (feloy) [inline chosen-docs/upstream/deploying-a-devfile-using-odo-on-IBM-Z-and-Power.md:20]: content should be "Devfile file reference"
and url "/file-reference/"

Comment (feloy) [reply chosen-docs/upstream/deploying-a-devfile-using-odo-on-IBM-Z-and-Power.md:20]: Also change the "next" section in `chosen-docs/upstream/deploying-a-devfile-using-odo.md` file so it points to this new document and not the file-reference anymore

**Comment by zhengxiaomei123** [comment]
Have rerun the script/sync.sh.

**Pull request merged**